x/ibc/02-client/simulation/decoder.go

```diff
@@ -4,29 +4,26 @@ import (
 	"bytes"
 	"fmt"
 
-	"github.com/cosmos/cosmos-sdk/codec"
 	"github.com/cosmos/cosmos-sdk/types/kv"
-	"github.com/cosmos/cosmos-sdk/x/ibc/02-client/exported"
+	"github.com/cosmos/cosmos-sdk/x/ibc/02-client/keeper"
 	host "github.com/cosmos/cosmos-sdk/x/ibc/24-host"
 )
 
 // NewDecodeStore returns a decoder function closure that unmarshals the KVPair's
 // Value to the corresponding client type.
-func NewDecodeStore(cdc *codec.Codec, kvA, kvB kv.Pair) (string, bool) {
+func NewDecodeStore(k keeper.Keeper, kvA, kvB kv.Pair) (string, bool) {
```

Comment:
Why did this signature have to change? Is this how other modules do it?

Reply:
yeah because it’s now the keeper that handles the encoding. Will do a follow-up PR to update the rest of the IBC submodules

Reply:
it has been updated to use an interface that is supplied only unmarshal methods for client and consensus state